components/molecules/Table.jsx

```diff
@@ -0,0 +1,218 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+// table components
+import Header from './TableHeader';
+import Heading from './TableHeading';
+import Body from './TableBody';
+import Row from './TableRow';
+import Data from './TableData';
+
+const sortDirection = {
+    ASC: 'asc',
+    DESC: 'desc'
+};
+
+class Table extends Component {
+    static getDerivedStateFromProps(nextProps) {
```

Comment:
We have to be careful with the new methods, this will probably be a minor update for semver, but the README should probably state the minimum React version needed to use the component. This then brings us a new issue, should we update the minimum peerDependency as well? I think so (shrug)

Reply:
Ya definitely, was thinking about this when I added it.

So I think there's a few things we can do which you touched upon:

1. Bump the minimum version of React required as a peer dependency
2. Use a polyfill to cover new methods but not introduce breaking changes to gumdrops: https://github.com/reactjs/react-lifecycles-compat
3. Keep the minimum React peerDep version as is, document minimum required versions (if different) in the component Readme
4. Don't use new features, yet.

I prefer option 1. I think most of our apps are already on 16? so the bump isn't a hard update.

Reply:
Yeah, let's go with the first one. We can't stop progress, let's just make sure we state the minimum react version needed for the component. I assume others will be unaffected.

Reply:
yeah, can we add something to the readme for this component that says you need 16.3? since it's a new component, no one will be using it yet, but we should warn people in case they see it and want to use it.